addons/payment/models/payment_acquirer.py

```diff
@@ -429,7 +429,7 @@ def _get_default_partner_country_id(self):
     # duplicate partner / transaction data to store the values at transaction time
     partner_id = fields.Many2one('res.partner', 'Partner', track_visibility='onchange')
     partner_name = fields.Char('Partner Name')
-    partner_lang = fields.Selection(_lang_get, 'Language', default='en_US')
```

Comment:
on [partner](https://github.com/odoo/odoo/blob/79bd2f3062b039d39b6f0f0259a9e8efae08acc8/odoo/addons/base/res/res_partner.py#L149), we use `default=lambda self: self.env.lang` Have you tried it?